packages/expo-notifications/src/NotificationsEmitter.ts

```diff
@@ -1,72 +1,38 @@
 import { Notification, NotificationResponse } from './NotificationsEmitter.types';
-
-interface NotificationReceivedEvent {
-  type: 'notificationReceived';
-  notification: Notification;
-}
-
-interface NotificationResponseEvent {
-  type: 'notificationResponseReceived';
-  response: NotificationResponse;
-}
-
-interface NotificationsDroppedEvent {
-  type: 'notificationsDropped';
-}
-
-export type NotificationEvent =
-  | NotificationReceivedEvent
-  | NotificationResponseEvent
-  | NotificationsDroppedEvent;
-
-export type NotificationListener = (notification: NotificationEvent) => void;
+export { Notification, NotificationResponse } from './NotificationsEmitter.types';
 
 // Web uses SyntheticEventEmitter
 const emitter = new EventEmitter(NativeModulesProxy.ExpoNotificationsEmitter);
+
 const didReceiveNotificationEventName = 'onDidReceiveNotification';
 const didDropNotificationsEventName = 'onNotificationsDeleted';
 const didReceiveNotificationResponseEventName = 'onDidReceiveNotificationResponse';
 
-type CompoundSubscription = Subscription & {
-  __subscriptions: Subscription[];
-};
+export function addNotificationReceivedListener(
+  listener: (event: Notification) => void
+): Subscription {
+  return emitter.addListener<Notification>(didReceiveNotificationEventName, listener);
+}
 
-export function addNotificationListener(listener: NotificationListener): Subscription {
-  const subscriptions = [
-    emitter.addListener<Notification>(didReceiveNotificationEventName, notification => {
-      listener({ notification, type: 'notificationReceived' });
-    }),
-    emitter.addListener<NotificationResponse>(didReceiveNotificationResponseEventName, response => {
-      listener({ response, type: 'notificationResponseReceived' });
-    }),
-    emitter.addListener<void>(didReceiveNotificationResponseEventName, () => {
-      listener({ type: 'notificationsDropped' });
-    }),
-  ];
+export function addNotificationsDroppedListener(listener: () => void): Subscription {
+  return emitter.addListener<void>(didDropNotificationsEventName, listener);
+}
 
-  return {
-    remove: () => {
-      for (let subscription of subscriptions) {
-        subscription.remove();
-      }
-    },
-    __subscriptions: subscriptions,
-  } as CompoundSubscription;
+export function addNotificationResponseReceivedListener(
+  listener: (event: NotificationResponse) => void
+): Subscription {
+  return emitter.addListener<NotificationResponse>(
+    didReceiveNotificationResponseEventName,
+    listener
+  );
 }
 
-export function removeNotificationSubscription(subscription: Subscription) {
-  if ('__subscriptions' in subscription) {
-    const compoundSubscription: CompoundSubscription = subscription;
-    for (let subscription of compoundSubscription.__subscriptions) {
-      emitter.removeSubscription(subscription);
-    }
-  } else {
-    emitter.removeSubscription(subscription);
-  }
+export function removeSubscription(subscription: Subscription) {
```

Comment:
is there any difference between calling this and `subscription.remove()`? if not, this seems redundant.

Reply:
True, I agree! This is also what James pointed out in https://github.com/expo/expo/pull/6577#discussion_r367219901

> ~removeTokenSubscription~ removePushTokenSubscription
> (although, is this function needed if developers can just call `subscription.remove()`?)

, to which I responded with https://github.com/expo/expo/pull/6779#issuecomment-575152833

> I agree developers can just call `subscription.remove()`, but eg. `EventEmitter.ts` and other event emitters (like `MediaLibrary`) implement this method anyway.
> 
> Maybe if we decide to remove this method from all event emitters it should be in a separate PR?

which has been approved of. Do you think we should remove this method now, or should we remove all of them immediately, or postpone the change until… some day?

Reply:
Gotcha. Sorry to make you justify this twice!! Let's leave it for now for consistency.

Reply:
This only proves that the review process is consistently good and thorough! 😃